__tests__/__snapshots__/plugin.test.js.snap

```diff
@@ -261,9 +261,54 @@ function Box() {
 }"
 `;
 
+exports[`babel plugin transforms nested functions correctly 1`] = `
+"var x = 5;
+var _worklet_8019770278350_init_data = {
+  code: \\"function inner(){const{x}=this._closure;return x;}\\",
+  location: \\"${ process.cwd() }/jest tests fixture\\"
+};
+var _worklet_8471062811730_init_data = {
+  code: \\"function f(){const{x,_worklet_8019770278350_init_data}=this._closure;var inner=function(){var _e=[new Error(),-2,-20];var _f=function _f(){return x;};_f._closure={x:x};_f.__initData=_worklet_8019770278350_init_data;_f.__workletHash=8019770278350;_f.__stackDetails=_e;return _f;}();return inner();}\\",
+  location: \\"${ process.cwd() }/jest tests fixture\\"
+};
+
+var f = function () {
+  var _e = [new Error(), -3, -20];
+
+  var _f = function _f() {
+    var inner = function () {
+      var _e = [new Error(), -2, -20];
+
+      var _f = function _f() {
+        return x;
+      };
+
+      _f._closure = {
+        x: x
+      };
+      _f.__initData = _worklet_8019770278350_init_data;
+      _f.__workletHash = 8019770278350;
+      _f.__stackDetails = _e;
+      return _f;
+    }();
+
+    return inner();
+  };
+
+  _f._closure = {
+    x: x,
+    _worklet_8019770278350_init_data: _worklet_8019770278350_init_data
+  };
+  _f.__initData = _worklet_8471062811730_init_data;
+  _f.__workletHash = 8471062811730;
+  _f.__stackDetails = _e;
+  return _f;
+}();"
+`;
+
 exports[`babel plugin transforms spread operator in worklets for arrays 1`] = `
-"var _worklet_3161057533258_init_data = {
-  code: \\"function foo(){const bar=[4,5];const baz=[1,...[2,3],...bar];}\\",
+"var _worklet_8509370056349_init_data = {
+  code: \\"function foo(){var bar=[4,5];var baz=[1].concat([2,3],bar);}\\",
```

Comment:
When I was working on #3143 I checked that it's safe to use spread operator in the worklet code. I assume that for some reason previously spread operator in worklets wasn't getting transformed but after the changes in this PR now it is. Do we know how this change affects the performance and memory usage?